anonym-ip.go

```diff
@@ -19,8 +20,46 @@ import (
 
 var (
 	ErrBrokenIP = errors.New("broken IP address")
+	ErrKeySize  = errors.New("illegal key size")
 )
 
+const BlockSize = 16
+
+// Ipcipher implements cipher.Block interface
+type Ipcipher struct {
+	key [BlockSize]byte
+}
+
+func (c *Ipcipher) BlockSize() int { return BlockSize }
+
+func NewCipher(key []byte) (cipher.Block, error) {
+	switch k := len(key); k {
+	default:
+		return nil, ErrKeySize
+	case BlockSize:
+		break
+	}
+	var a [BlockSize]byte
+	for i := range key {
+		a[i] = key[i]
```

Comment:
minor: if you use range you could use both values:
```suggestion
	for i,v := range key {
		a[i] = v
```
you could also use copy:
```suggestion
	copy(a[:], key)
```
Not sure what would be faster (the copy version creates a tmp slice over the array).